ckanext/datastore/backend/postgres.py

```diff
@@ -1921,6 +2041,21 @@ def delete(self, context, data_dict):
 
             trans.commit()
             return _unrename_json_field(data_dict)
+        # (canada fork only): foreign keys
+        # TODO: upstream contrib!!
+        except (IntegrityError, ProgrammingError, InternalError) as e:
+            trans.rollback()
+            if e.orig.pgcode == _PG_ERR_CODE['row_referenced_constraint'] or e.orig.pgcode == _PG_ERR_CODE['table_referenced_constraint']:
+                # FIXME: better error message??
+                raise ValidationError({
+                    'foreign_constraints': ['Cannot delete records or table'
+                                            ' because of a reference to another table.'],
```

Comment:
Might be fine for PD but foreign keys would allow user A to prevent user B from dropping or modifying data in a table user B owns. They could even be used to extract information from a private resource.

Not immediately obvious how to solve this. Maybe we need a way to declare when we're creating a table that an index can be used as a fk in other tables?

Reply:
flagged issue for upstream.